pkg/gatewayserver/io/ws/ws.go

```diff
@@ -170,7 +170,7 @@ func (s *srv) handleTraffic(w http.ResponseWriter, r *http.Request) (err error)
 		ctx          = r.Context()
 		eps          = s.formatter.Endpoints()
 		missedPongs  = noPongReceived
-		pongCh       = make(chan struct{}, 1)
+		pongCh       = make(chan string, 1)
```

Comment:
Nit: We can use a byte channel instead and cast the data at the sender.

Reply:
I don't fully see the gain here - someone has to cast, and both casts have to copy the data (because you don't want the byte slice to edit your original string, and you don't want your slice edits to be visible in your new string).